### ISSUE TITLE
adios2-config installation failure

### ISSUE THREAD

**Issue by meerfrau** [ornladios/ADIOS2#3669]
We ask the community to always report bugs. Please open a new issue if some other comes up in the conversation. It will help us stay on topic and manage tasks more efficiently.

**Describe the bug**
After the compilation and main installation have finished, it fails on the last steps:
```
-- Installing: ./pkg/adios2/usr/lib/cmake/adios2/adios2-config.cmake
/tmp/tmp.WQe96LCPvH ~/aur/__rebuilds/adios2/src/build
Configuring adios2-config dummy CMake project
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/mpicxx - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:9 (find_package):
  By not providing "Findadios2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "adios2", but
  CMake did not find one.

  Could not find a package configuration file provided by "adios2" with any
  of the following names:

    adios2Config.cmake
    adios2-config.cmake

  Add the installation prefix of "adios2" to CMAKE_PREFIX_PATH or set
  "adios2_DIR" to a directory containing one of the above files.  If "adios2"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
Failed
CMake Error at cmake/install/post/cmake_install.cmake:49 (message):
  generate-adios2-config.sh returned 1, should be 0
Call Stack (most recent call first):
  cmake_install.cmake:58 (include)
```

**To Reproduce**
```
  cmake -B build -S ${pkgname^^}-${pkgver} \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_SKIP_RPATH=ON \
    -DADIOS2_USE_EXTERNAL_DEPENDENCIES=ON \
    -DADIOS2_HAVE_HDF5_VOL=OFF \
    -DADIOS2_BUILD_EXAMPLES=OFF #\
    #-DMGARD_INCLUDE_DIR=/usr/include/mgard
    #-DADIOS2_HAVE_HDF5_VOL needs hdf5-openmpi
  make VERBOSE=1 -C build
```
`
  make -C build DESTDIR="${pkgdir}" install`

**Expected behavior**
 adios2-config dummy CMake project should not make any problems

**Desktop (please complete the following information):**
 - OS/Platform: ArchLinux
 - Build: gcc 13.1.1, cmake 3.26.4

**Comment by vicentebolea** [comment]
Which ADIOS2 version are you using?

**Comment by meerfrau** [comment]
current git master

**Comment by vicentebolea** [comment]
related #3473

**Comment by vicentebolea** [comment]
@meerfrau we want to get rid of this adios2-config shell script soon. In the meanwhile you can disable the generation of it and its testing (which fails here) with: `-DADIOS2_INSTALL_GENERATE_CONFIG=OFF`

**Comment by meerfrau** [comment]
@vicentebolea This would indeed work!

**Comment by vicentebolea** [comment]
This might have been resolved at #3760

**Comment by vicentebolea** [comment]
Closing this since we cannot reproduce this anymore. Please re-open if you observe this again
